src/formatter-settings/index.ts

```diff
@@ -26,5 +31,65 @@ export class JavaFormatterSettingsEditorProvider implements vscode.CustomTextEdi
         };
         const resourceUri = this.context.asAbsolutePath("./out/assets/formatter-settings/index.html");
         webviewPanel.webview.html = await loadTextFromFile(resourceUri);
+        this.exampleKind = ExampleKind.INDENTATION_EXAMPLE;
+        webviewPanel.webview.onDidReceiveMessage(async (e) => {
+            switch (e.command) {
+                case "onWillInitialize": {
+                    // fake provider: use default values
+                    webviewPanel.webview.postMessage({
+                        command: "loadProfileSetting",
+                        setting: Array.from(this.supportedProfileSettings.values()),
+                    });
+                    webviewPanel.webview.postMessage({
+                        command: "loadVSCodeSetting",
+                        setting: Array.from(this.supportedVSCodeSettings.values()),
+                    });
+                    this.formatWithProfileSettings(webviewPanel);
+                    break;
+                }
+                case "onWillChangeExampleKind": {
+                    this.exampleKind = e.exampleKind;
+                    this.formatWithProfileSettings(webviewPanel);
+                    break;
+                }
+                case "onWillChangeSetting": {
+                    // fake provider: replace the setting directly 
```

Comment:
what's the meaning of fake provider?

Reply:
Here we just change the setting in memory instead of modifying the file content. I'll change the description to make it clear.